Add html table to provide a protocol overview

ixxie opened the pull request:
The protocol specification is complex, and I found that as an implementer I benefited from a visual overview.

You can [preview the table here](https://github.com/ixxie/tus-resumable-upload-protocol/blob/protocol-table/protocol.md#overview).

p.s. I'm a new contributor to the [tuspyserver](https://github.com/edihasaj/tuspyserver) Python / FastAPI server implementation; I'm currently [refactoring](https://github.com/edihasaj/tuspyserver/pull/64) the code and examples to make it easier to maintain, and I hope to add the missing extensions soon.

ixxie committed:
Add html table to provide a protocol overview

The protocol specification is complex, and I found that as an
implementer I benefited from a visual overview.

smatsson commented:
I like the idea behind this but I think it's more suited as an appendix than "straight in your face" the first thing you get to when reading the spec. Also not a fan of all the emojis as there aren't any in the rest of the spec. What do you think @Acconut ?

ixxie commented:
> I like the idea behind this but I think it's more suited as an appendix than "straight in your face" the first thing you get to when reading the spec. Also not a fan of all the emojis as there aren't any in the rest of the spec. What do you think @Acconut ?

I was hesitating also about the position. Its quite large as a header.

Acconut commented:
Thank you for this proposal! I can imagine that such table is helpful by providing an overview of the different header fields and requests methods that are in use. We have to be careful with the wording and make sure that there are no discrepancies with the remaining text in the specification. For that it would be useful to label this table as informational-only, so that the normative text has precedence even if the table and text give conflicting information.

Other than that, I agree with @smatsson that it the table would be better placed in an appendix.

ixxie commented:
@smatsson @Acconut I now moved the table to an appendix and added a note to emphasize the section is non-normative.

As a little compromise, I added a sentence to the first section of the document that links to it. This emphasizes the non-normativity of that table but offers new-commers and accessible jump to the overview.
